test/yup-middleware.spec.ts

```diff
@@ -195,4 +198,125 @@ describe('express yup middleware', () => {
       })
     })
   })
+
+  describe('when using the request as an Yup context to cross validate', () => {
+    const shouldBeOfTypeAccoringTo = (source) =>
+      function shouldBeEven(this: Yup.TestContext, numberToValidate: any) {
+        const mod = numberToValidate % 2
+        const type = this.options.context['req'][source].type
+
+        if (!type) {
+          return false
+        }
+
+        if (type === 'even') return mod === 0
+        if (type === 'odd') return mod !== 0
+
+        return false
+      }
+
+    it('returns a bad request error using a query value within a param validation', async () => {
+      const schemaValidator: ExpressYupMiddlewareInterface = {
+        schema: {
+          params: {
+            yupSchema: Yup.object().shape({
+              numberToValidate: Yup.string().test({
+                message: 'Check if your number correspond with the type given',
+                test: shouldBeOfTypeAccoringTo('query'),
+              }),
+            }),
+          },
+        },
+      }
+
+      const app = createAppWithPath({
+        path: '/test/:numberToValidate',
+        middleware: expressYupMiddleware({ schemaValidator }),
+      })
+
+      agent = request(app)
+
+      const { body } = await agent.get('/test/1?type=even').expect(400)
+      await agent.get('/test/1?type=odd').expect(200)
+      await agent.get('/test/2?type=even').expect(200)
+
+      expect(body).toStrictEqual({
+        errors: {
+          params: [
+            {
+              propertyPath: 'numberToValidate',
+              message: 'Check if your number correspond with the type given',
+            },
+          ],
+        },
+      })
+    })
+
+    it('returns a bad request error using a param value within a body validation', async () => {
```

Comment:
returns a bad request error when the number sent in the payload does not match the type param